delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/model/CompositeViewSource.scala

```diff
@@ -0,0 +1,106 @@
+    */
+  def resourceTag: Option[String]
+
+  /**
+    * @return whether to consider deprecated resources for indexing
+    */
+  def includeDeprecated: Boolean
+}
+
+object CompositeViewSource {
+
+  /**
+    * A source for the current project.
+    *
+    * @param id                 the id of the source.
+    * @param resourceSchemas    the set of schemas considered for indexing; empty implies all
+    * @param resourceTypes      the set of resource types considered for indexing; empty implies all
+    * @param resourceTag        an optional tag to consider for indexing; when set, all resources that are tagged with
+    *                           the value of the field are indexed with the corresponding revision
+    * @param includeDeprecated  whether to consider deprecated resources for indexing
+    */
+  final case class ProjectSource(
+      id: Iri,
+      resourceSchemas: Set[Iri],
+      resourceTypes: Set[Iri],
+      resourceTag: Option[String],
+      includeDeprecated: Boolean
+  ) extends CompositeViewSource
+
+  /**
+    * A cross project source.
+    *
+    * @param id                 the id of the source.
+    * @param resourceSchemas    the set of schemas considered for indexing; empty implies all
+    * @param resourceTypes      the set of resource types considered for indexing; empty implies all
+    * @param resourceTag        an optional tag to consider for indexing; when set, all resources that are tagged with
+    *                           the value of the field are indexed with the corresponding revision
+    * @param includeDeprecated  whether to consider deprecated resources for indexing
+    * @param project            the project to which source refers to
+    * @param identities         the identities used to access the project
+    */
+  final case class CrossProjectSource(
+      id: Iri,
+      resourceSchemas: Set[Iri],
+      resourceTypes: Set[Iri],
+      resourceTag: Option[String],
+      includeDeprecated: Boolean,
+      project: ProjectRef,
+      identities: Set[Identity]
+  ) extends CompositeViewSource
+
+  /**
+    * A remote project source
+    *
+    * @param id                 the id of the source
+    * @param resourceSchemas    the set of schemas considered for indexing; empty implies all
+    * @param resourceTypes      the set of resource types considered for indexing; empty implies all
+    * @param resourceTag        an optional tag to consider for indexing; when set, all resources that are tagged with
+    *                           the value of the field are indexed with the corresponding revision
+    * @param includeDeprecated  whether to consider deprecated resources for indexing
+    * @param endpoint           the endpoint used to access the source
+    * @param token              the optional access token used to connect to the endpoint
+    */
+  final case class RemoteProjectSource(
+      id: Iri,
+      resourceSchemas: Set[Iri],
```

Comment:
Maybe it should be `ResourceRef` to support a schema in a certain revision? But probably don't add it yet, since it will make things more complicated